### PR TITLE
Update Typescript migration documentation for Merging

### PR DESCRIPTION
**Pull request by ong6**
**What is the purpose of this pull request?**

- [x] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Update the dev docs regarding typescript migration to use rebase and merge instead of normal merge.

**Anything you'd like to highlight/discuss:**

**Testing instructions:**

**Proposed commit message: (wrap lines at 72 characters)**
Update Typescript migration documentation for Merging

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes

### CODE REVIEW

File: docs/devGuide/development/migratingToTypeScript.md [hunk 1]
```diff
@@ -149,13 +149,18 @@ Combining the two steps into one - that is, adapting files immediately after ren
...
 
-### Why do we need a normal merge commit
+### Why do we need a rebase and merge commit
 
 Even if the migration developer has kept the history intact through the separate "Rename" and "Adapt" commits, this is only intact at their working branch. At the end of the day, the totality of the changes in the working branch is compared against the target branch, in which the same consideration would apply.
 
 If we do the usual squash commit, the changes from the two commits are combined into a new commit and only that commit will be pushed into the target branch. The original two commits are omitted, therefore the history of the working branch that we have tried to keep intact is stripped away.
 
-The normal merge commit also creates another commit (the merge commit), but the merge process interleaves the commits from the working branch to the target branch, therefore retaining the change history.
+The reason we use a rebase operation is because it updates the base branch of a
```
Comment (raysonkoh) [inline docs/devGuide/development/migratingToTypeScript.md:158]: I think the last sentence is fine. I think there's no need to define what a rebase is.

**Review by tlylt — COMMENTED** [review]
Would be nice if we can add some screenshots to show the exact steps and the expected outcome, so as to reduce unnecessary mental stress.

File: docs/devGuide/development/migratingToTypeScript.md [hunk 2]
```diff
@@ -95,6 +95,31 @@ We have decided on structuring the commits as two commits: a "Rename" commit and
...
+
+To append a new commit onto the already existing commit you can do the following: 
+1. Make the changes you want to add to the second commit
+2. Stage the changes using `git add`
+3. Run `git commit --amend`
+4. This will open your default text editor, where you can edit the commit message for the second commit to reflect the new changes you have made
+5. Save and close the text editor. This will replace the second commit with the updated changes.
+
+<box type="warning">
+It's important to note that amending a commit changes its commit hash, which means that if you've already pushed the original commit to a remote repository, you'll need to force push your changes.
```
Comment (jovyntls) [inline docs/devGuide/development/migratingToTypeScript.md:112]:
```suggestion
Amending a commit changes its commit hash, which means that if you've already pushed the original commit to a remote repository, you'll need to force push your changes.
```

File: docs/devGuide/development/migratingToTypeScript.md [hunk 3]
```diff
@@ -95,6 +95,31 @@ We have decided on structuring the commits as two commits: a "Rename" commit and
 
 You are now ready to create a pull request for the changes to the repository.
 
+### (Optional) Step 4: Changing the commit
+
+While refactoring you may encounter scenarios where you might have to amend your code after a commit. There are 2 common scenarios which we will cover. 
+
+##### Scenario 1 - you have 2 commits and want to add on to the second commit:
```
Comment (jovyntls) [inline docs/devGuide/development/migratingToTypeScript.md:102]: nit: can we exclude these level-5 headers from the page nav? The headers are quite long and may clutter up the page nav.
Or maybe we can put it in a panel since this section is optional?

File: docs/devGuide/development/migratingToTypeScript.md [hunk 4]
```diff
@@ -95,6 +95,31 @@ We have decided on structuring the commits as two commits: a "Rename" commit and
...
+4. This will open your default text editor, where you can edit the commit message for the second commit to reflect the new changes you have made
+5. Save and close the text editor. This will replace the second commit with the updated changes.
+
+<box type="warning">
+It's important to note that amending a commit changes its commit hash, which means that if you've already pushed the original commit to a remote repository, you'll need to force push your changes.
+</box>
+
+##### Scenario 2 - you have 3 commits and want to make it into 2 commits:
+
+1. Run `git rebase -i HEAD~3` to start an interactive rebase session. This will show a list of the three most recent commits in your text editor.
```
Comment (jovyntls) [inline docs/devGuide/development/migratingToTypeScript.md:117]:
```suggestion
1. Run `git rebase -i master` to start an interactive rebase session. This will show a list of the recent commits in your text editor.
```

Rebasing on master can allow this command to be used regardless of how many commits there are! Also bonus of making the rebase commit (for merging into master) easier later on.

File: docs/devGuide/development/migratingToTypeScript.md [hunk 5]
```diff
@@ -95,6 +95,31 @@ We have decided on structuring the commits as two commits: a "Rename" commit and
...
+2. Stage the changes using `git add`
+3. Run `git commit --amend`
+4. This will open your default text editor, where you can edit the commit message for the second commit to reflect the new changes you have made
+5. Save and close the text editor. This will replace the second commit with the updated changes.
+
+<box type="warning">
+It's important to note that amending a commit changes its commit hash, which means that if you've already pushed the original commit to a remote repository, you'll need to force push your changes.
+</box>
+
+##### Scenario 2 - you have 3 commits and want to make it into 2 commits:
```
Comment (jovyntls) [inline docs/devGuide/development/migratingToTypeScript.md:115]:
```suggestion
##### Scenario 2 - you have >2 commits and want to make it into 2 commits:
```

File: docs/devGuide/development/migratingToTypeScript.md [hunk 6]
```diff
@@ -149,13 +174,12 @@ Combining the two steps into one - that is, adapting files immediately after ren
...
 
 Even if the migration developer has kept the history intact through the separate "Rename" and "Adapt" commits, this is only intact at their working branch. At the end of the day, the totality of the changes in the working branch is compared against the target branch, in which the same consideration would apply.
 
-If we do the usual squash commit, the changes from the two commits are combined into a new commit and only that commit will be pushed into the target branch. The original two commits are omitted, therefore the history of the working branch that we have tried to keep intact is stripped away.
-
-The normal merge commit also creates another commit (the merge commit), but the merge process interleaves the commits from the working branch to the target branch, therefore retaining the change history.
+If we do the usual squash commit, the changes from the two commits are combined into a new commit and only that commit will be pushed into the target branch. The original two commits are omitted, therefore the history of the working branch that we have tried to keep intact is stripped away. This results in a linear history, as if the changes in the working branch were made
```
Comment (jovyntls) [inline docs/devGuide/development/migratingToTypeScript.md:181]: I don't think the point about linear history is necessarily relevant here - our repo history is quite linear because of the squash and merge strategy so using a rebase over squash+merge doesn't add much linearity. Maybe can we revert this change?

**Review by jovyntls — COMMENTED** [review]
Thanks for the additions @ong6 ! Added some comments

File: docs/devGuide/development/migratingToTypeScript.md (hunk 3 above)
Comment (ong6) [reply docs/devGuide/development/migratingToTypeScript.md:102]: Good idea, ill put this into a panel!

**Review by jovyntls — APPROVED** [review]
LGTM! Thanks for the additions @ong6

File: docs/devGuide/development/migratingToTypeScript.md (hunk 6 above)
Comment (tlylt) [reply docs/devGuide/development/migratingToTypeScript.md:194]: @ong6 this is not reverted yet?

**Review by tlylt — APPROVED** [review]
LGTM
